### DISCUSSION TITLE
Basic Authentication with Bearer Token

### DISCUSSION THREAD

**Discussion by junioriwata** [grafana/grafana-infinity-datasource#522] [Specific APIs]
Hi, @yesoreyeram, i need help
How do I resolve over an external API that generates a 30 minute temporary bearer token? How to renew this token automatically?

![image](https://user-images.githubusercontent.com/124163068/216058353-3c75e31b-2b44-4cb7-9dd4-c251ec592c3a.png)

**Comment by szvieira** [comment]
Did you solve that?

In my case, I've used other plugin that required the same thing. To solve this 30 min expiration I wrote a playbook and put it on cron to run the request that gives me the Bearer Token and in another task I updated the data source on Grafana changing the token that had expired.

My question is if this plugin have something like this natively
